CONTRIBUTING.md

```diff
@@ -1,48 +1,103 @@
-    * Even if someone else is doing something similar, there may still be opportunity to contribute. Comment on the draft PR with your questions and suggestions.
+* All project files, especially source files need to have a compatible license with the project. See [LICENSE](LICENSE).
+* There is no written standard on formatting and coding guidelines yet so the current requirement is to make your code look and feel like the rest of the repository and respect the styling decisions taken therein (for example brace position). Clang formatting has been included in the GitHub Actions and should help with this.
+* All changes to the firmware have to be tested on a best effort basis to make sure they work as expected and don't break any existing functionality before submitting as ready to merge. This does not apply to Draft Pull requests.
+* All changes need to be compatible with all available hardware variants, this currently includes OLED and 7-Segment.
+* All changes need to be compatible with the currently official toolchain as described in the [Readme](README.md).
+* If a CI system has been established the acceptance of the CI system is also a requirement (comprising probably of compilation and (unit) testing).
 
-* Issues and Draft PRs are *not* the place for feature requests by non-developers.
+### Application specific
 
-### Low-effort Pull Requests
+* Pull requests that change how users can interact with the device or massively alter system performance (> 3% permanent cycle load) require either.
+    * A runtime configuration setting that allows to enable or disable the feature/change in behavior, see documentation on adding optional feature settings (Pull request #56).
+    * Or if a runtime setting is not possible a preprocessor switch that allows creating firmware without the change.
+* Changes that massively increase image size (> 5% of total memory) also require a preprocessor switch starting with "FEATURE_" so they can be enabled or disabled.
+* If the Pull request requires changes in structure of user files (e.g. project/synth or other xml files) or flash configuration.
+    * It must ensure that files or flash configuration created with the official firmware or previous community releases stay compatible or are automatically upgraded (upward compatibility).
+    * If possible files and flash configuration created with the changes of the Pull request can be used with older firmwares not having the change (downward compatibility). Older firmwares must not break if a newer configuration was stored on the device.
+* If the Pull request changes end user behavoir or introduces new features a new entry in the [CommunityFeatures.md](CommunityFeatures.md) file needs to be created in the preexisting style describing the feature and it's options as a small manual to users. This includes all runtime and compile time flags which shall be named in respective sections.
```

Comment:
I like this idea. perhaps create a stub of this file in this PR as well with some structure to encourage feature PR:s to fill it in?

Reply:
I added a https://github.com/SynthstromAudible/DelugeFirmware/pull/56/commits/6d1d7f96801c41bd15375fbf72fe117e2e44cf77 , hope it fits your expectations :)